modules/data.atmosphere/R/load.cfmet.R

```diff
@@ -51,11 +51,18 @@ load.cfmet <- function(met.nc, lat, lon, start.date, end.date) {
 
   all.dates <- data.frame(index = seq_along(time.idx), date = date)
   
-  if (start.date + lubridate::days(1) < min(all.dates$date)) {
-   PEcAn.logger::logger.severe("run start date", start.date, "before met data starts", min(all.dates$date))
+  if (nrow(all.dates) > 1) {
+    delta <- stats::median(diff(all.dates$date), na.rm = TRUE)
+  } else {
+    delta <- as.difftime(1, units = "hours")  
   }
```

Comment:
I think one-timepoint files will be very rare; instead of testing for them as part of the happy path, handling them as part of a more general exception feels clearer to me:

```suggestion
  delta <- stats::median(diff(all.dates$date), na.rm = TRUE)
  if (is.na(delta)) {
    # probably only happens with a one-line met file
    # fall back to requiring exact match
    delta <- 0
  }
```

Reply:
Would hardcoding 0 as the fallback for delta be correct in this context? Since the timestep is likely meant to represent a duration, would it be more appropriate to default to as.difftime(1, units = "hours") instead?

Reply:
I agree that delta represents a duration, but if we can't discern it from the file then we don't even know what unit to use -- it could be seconds, days, months, etc.

Falling back to 0 will give essentially the current behavior: It will fail if the last timestamp in the file isn't >= the requested end time. In the one-timestep case that means the user can get past this by passing identical start and end times, which seems reasonable to me -- and maybe even a useful check on whether they realized they were asking for only one timepoint.

Reply:
> but if we can't discern it from the file then we don't even know what unit to use -- it could be seconds, days, months, etc.

yes this makes sense